test/parallel/test-zlib-write-after-close.js

```diff
@@ -20,15 +20,24 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-const common = require('../common');
-const zlib = require('zlib');
 
-zlib.gzip('hello', common.mustCall(function(err, out) {
-  const unzip = zlib.createGunzip();
-  unzip.close(common.mustCall());
-  unzip.write('asd', common.expectsError({
-    code: 'ERR_STREAM_DESTROYED',
-    name: 'Error',
-    message: 'Cannot call write after a stream was destroyed'
-  }));
-}));
+require('../common');
+
+const zlib = require('node:zlib');
+const assert = require('node:assert');
+const { test } = require('node:test');
+
+test('szlib should not allow writing after close', async () => {
+  const { promise, resolve } = Promise.withResolvers();
+  zlib.gzip('hello', function() {
+    const unzip = zlib.createGunzip();
+    unzip.close(resolve);
+    unzip.write('asd', function(err) {
+      assert.strictEqual(err.code, 'ERR_STREAM_DESTROYED');
+      assert.strictEqual(err.name, 'Error');
+      assert.strictEqual(err.message, 'Cannot call write after a stream was destroyed');
+      resolve();
```

Comment:
This calls resolve twice, shouldn't we have 2 different promises and an `await Promise.all(...)`?

Reply:
Yeah, if we want to ensure that the write callback was invoked we should use a `mock.fn()` to wrap the asserts and check the call count after the `await promise` below.